### PR TITLE
CI: Address linting errors in flake8 >= 3.8.1

### PR DESCRIPTION
**Pull request by mgmarino**
xref #34150

This partially addresses the above issue by:

- Ignoring E741
- Fixing other linting errors

The issue with flake8-rst is handled temporarily in #34151, and the build here will fail the flake8-rst part until that is merged in.

### CODE REVIEW

File: pandas/core/internals/concat.py [hunk 1]
```diff
@@ -443,7 +443,10 @@ def _is_uniform_join_units(join_units: List[JoinUnit]) -> bool:
     #  cannot necessarily join
     return (
         # all blocks need to have the same type
-        all(type(ju.block) is type(join_units[0].block) for ju in join_units)
+        all(
+            type(ju.block) is type(join_units[0].block)  # noqa: E721
```
Comment (jreback) [inline pandas/core/internals/concat.py:447]: instead use `instance`

**Review by jreback — CHANGES_REQUESTED** [review]
also merge master

File: pandas/core/internals/concat.py (hunk 1 above)
Comment (mgmarino) [reply pandas/core/internals/concat.py:447]: Thanks, I wasn't sure in this in this case if the *exact* type was actually important.  Will update.

**Comment by jreback** [comment]
great, does this complete the flake issues? e.g. can we unpin after this one?

**Comment by mgmarino** [comment]
> great, does this complete the flake issues? e.g. can we unpin after this one?

It fixes all the issues with flake8 in the code, but it unfortunately doesn't fix the issue with flake8-rst [here](kataev/flake8-rst#22).  That's a problem with the `flake8-rst` tool itself.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok thanks.

**Comment by simonjayhawkins** [comment]
@MeeseeksDev backport to 1.0.x

**Comment by simonjayhawkins** [comment]
@jorisvandenbossche will start backporting this or could pin flake8 on 1.0.x? lmk

File: pandas/core/internals/concat.py [hunk 2]
```diff
@@ -443,7 +443,7 @@ def _is_uniform_join_units(join_units: List[JoinUnit]) -> bool:
     #  cannot necessarily join
     return (
         # all blocks need to have the same type
-        all(type(ju.block) is type(join_units[0].block) for ju in join_units)
+        all(isinstance(ju.block, type(join_units[0].block)) for ju in join_units)
```
Comment (jorisvandenbossche) [inline pandas/core/internals/concat.py:446]: This change is actually not equivalent. We have both ExtensionBlock as other Block types that are subclasses from ExtensionBlock.

**Comment by jorisvandenbossche** [comment]
> @jorisvandenbossche will start backporting this or could pin flake8 on 1.0.x? lmk

I would personally not care about backporting style related issues. Just pinning flake8 seems easier (or letting the lint build fail ..)

File: pandas/core/internals/concat.py (hunk 2 above)
Comment (mgmarino) [reply pandas/core/internals/concat.py:446]: @jorisvandenbossche Thanks, in the first iteration of the PR, I actually retained the exact type check and opted for a `#  noqa`.  Should this be a new issue?  We could also require this behavior in a test.

Comment (jorisvandenbossche) [reply pandas/core/internals/concat.py:446]: I was actually just looking at this. In the meantime it also seems that the location where this `_is_uniform_join_units` is used isn't that strict anymore about the exact block types (due to https://github.com/pandas-dev/pandas/pull/33486), so it should be fine as is.

**Comment by simonjayhawkins** [comment]
> > @jorisvandenbossche will start backporting this or could pin flake8 on 1.0.x? lmk
> 
> I would personally not care about backporting style related issues. Just pinning flake8 seems easier (or letting the lint build fail ..)

yep. not backporting this at this time.

**Pull request merged**